subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -2097,6 +2109,47 @@ static isoal_status_t isoal_tx_framed_event_prepare_handle(isoal_source_handle_t
 		err = isoal_tx_try_emit_pdu(source, true, PDU_BIS_LLID_FRAMED);
 	}
 
+	if (pp->mode != ISOAL_PRODUCTION_MODE_DISABLED) {
+		/* BT Core V5.3 : Vol 6 Low Energy Controller :
+		 * Part G IS0-AL:
+		 *
+		 * 2 ISOAL Features :
+		 * Padding is required when the data does not add up to the
+		 * configured number of PDUs that are specified in the BN
+		 * parameter per CIS or BIS event.
+		 *
+		 * There is some lack of clarity in the specifications as to why
+		 * padding PDUs should be used as opposed to null PDUs. However
+		 * if a payload is not available, the LL must default to waiting
+		 * for the flush timeout before it can proceed to the next
+		 * payload.
+		 *
+		 * This means a loss of retransmission capacity for future
+		 * payloads that could exist. Sending padding PDUs will prevent
+		 * this loss while not resulting in additional SDUs on the
+		 * receiver. However it does incur the allocation and handling
+		 * overhead on the transmitter.
+		 *
+		 * As an interpretation of the specification, padding PDUs will
+		 * only be released if an SDU has been received in the current
+		 * event.
+		 */
+		if (pp->payload_number > first_event_payload) {
+			release_padding = true;
+		}
+	}
+
+	if (release_padding) {
+		while (!err && !err_alloc && (pp->payload_number < last_event_payload + 1)) {
+			err_alloc = isoal_tx_allocate_pdu(source, NULL);
+
+			err = isoal_tx_try_emit_pdu(source, true, PDU_BIS_LLID_FRAMED);
+		}
+	}
+
+	/* Not possible to recover if allocation or emit fails here*/
+	LL_ASSERT(!err && !err_alloc);
```

Comment:
```suggestion
	LL_ASSERT(err || err_alloc);
```

Reply:
I think there is a negation missing in the suggestion but I get the point.